lib/loaders/api/loader_interface.js

```diff
@@ -0,0 +1,32 @@
+import { toKey } from "lib/helpers"
+
+const encodeStaticPath = (path, globalParams, params) => {
+  return toKey(path, Object.assign({}, globalParams, params))
+}
+
+const encodeDynamicPath = (pathGenerator, globalParams, id, params) => {
+  return encodeStaticPath(pathGenerator(id), globalParams, params)
+}
+
+/**
+ * This implements the common interface for producing data loaders for each api/path, regardless of authentication.
+ *
+ * If `path` is a string, then the loader will accept only query params as its parameters. For example:
+ *
+ *    const artworksLoader = gravityLoader("artworks")
+ *    artworksLoader({ page: 42 })
+ *
+ * If `path` is a function, then the loader accepts an ID _and_ query params as its parameters. For example:
+ *
+ *    const artistArtworksLoader = gravityLoader(id => `artist/${id}/artworks`)
```

Comment:
I think the function is the better way now that I think more about it, i was just confused initially.

Yea, having params that may or may not be interpolated (and thus removed from the params sent to Grav) seems potentially _more_ confusing, I'm down for the function style.